core/client-internal/src/test/java/alluxio/client/netty/NettyRemoteBlockWriterTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.netty;
+
+import static org.hamcrest.core.StringStartsWith.startsWith;
+
+import alluxio.client.ClientContext;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+/**
+ * Unit tests for @{link NettyRemoteBlockWriter}.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ClientContext.class, NettyRemoteBlockWriter.class})
+public final class NettyRemoteBlockWriterTest {
+
+  private static final byte[] BYTES = new byte[]{(byte) 0xe0, 0x4f};
+  private static final int OFFSET = 10;
+  private static final int LENGTH = 20;
+  private static final long BLOCK_ID = 100L;
+  private static final long SESSION_ID = 200L;
+
+  private InetSocketAddress mInetSocketAddress;
+  private NettyRemoteBlockWriter mNettyRemoteBlockWriter;
+
+  /**
+   * Sets up the constructors before a test run.
+   */
+  @Before
+  public void before() throws IOException {
+    mInetSocketAddress = ClientContext.getMasterAddress();
+    mNettyRemoteBlockWriter = PowerMockito.spy(new NettyRemoteBlockWriter());
+    mNettyRemoteBlockWriter.open(mInetSocketAddress, BLOCK_ID, SESSION_ID);
+  }
+
+  /**
+   * Sets a rule for {@link IOException} exception.
```

Comment:
`The exception expected to be thrown` (no need to constrain it to `IOException` only here)